### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by dada-dell-emc**
**Change log description**
Added end to end integration test case for PSC.
**Purpose of the change**
Test PSC functionality end to end integrating with Pravega using ingestion files no mock.
**What the code does**
It start pravega related services at the start of test setup, after that as a part of test case **testRawFile**  it copies test ingestion files into respective directory from where PSC reads it and sends data to pravega. Test case checks that after 40-50 secs from start does database updates with completed files, if it does, then it waits another 50 secs to execute delete completed files tasks, once that wait is over test case checks completed files records from DB got deleted or not. This tests end to end PSC functionality.
**How to verify it**
Run all or particular test case(i.e. PravegaSensorCollectorIntegrationTests.testRawFile), it passes every time with in 1.5-2 mins.
**What tests you have run. (Manually, automated whatever). How can someone else test this?**
Follow steps mentioned above.

### CODE REVIEW

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java [hunk 1]
```diff
@@ -60,9 +69,91 @@ public void tearDown() {
...
+                        Assertions.assertEquals("Hello World.", event);
+                    }
+                } catch (ReinitializationRequiredException e) {
+                    //There are certain circumstances where the reader needs to be reinitialized
+                    e.printStackTrace();
+                }
+                System.out.format("No more events from %s/%s%n", scope, streamName);
+            }
+
+            Thread.sleep(50000);
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java:134]: Is this sleep intentional ? why 50K?

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java [hunk 2]
```diff
@@ -60,9 +69,91 @@ public void tearDown() {
 
     }
 
+    @Test
+    @Timeout(value = 3, unit = TimeUnit.MINUTES)
+    @Order(1)
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java:74]: Is there a way to not use order?

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java [hunk 3]
```diff
@@ -60,9 +69,91 @@ public void tearDown() {
...
+            copyHelloWorldFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        final DeviceDriverManager deviceDriverManager = new DeviceDriverManager(properties);
+        Service startService = deviceDriverManager.startAsync();
+        try {
+            startService.awaitRunning(Duration.ofSeconds(30));
+            Thread.sleep(25000);
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java:86]: Is this sleep necessary? Why 25K?

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java [hunk 4]
```diff
@@ -60,9 +69,91 @@ public void tearDown() {
...
+
+            final String readerGroup = UUID.randomUUID().toString().replace("-", "");
+            final ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                    .stream(Stream.of(scope, streamName))
+                    .build();
+            try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI)) {
+                readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+            }
+
+            try (EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope,
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java:111]: Please refactor by extracting validation code in its own method. Also makes it reusable.

**Review by sachin-j-joshi — COMMENTED** [review]
looks good overall. Left some comments regarding the order and sleeps.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        final Connection connection = SQliteDBUtility.createDatabase(properties.get("PRAVEGA_SENSOR_COLLECTOR_RAW1_DATABASE_FILE"));
+        final TransactionStateDB state = new TransactionStateSQLiteImpl(connection, null);
+
+        try {
+            List<FileNameWithOffset> completedFiles = state.getCompletedFileRecords();
+            Assertions.assertEquals(1, completedFiles.size());
+
+            validateStreamData(controllerURI, scope, streamName, new String(Files.readAllBytes(Paths.get("../parquet-file-sample-data/test_file/hello-world.parquet"))));
+
+            Thread.sleep(50000);
```
Comment (abhinb) [inline pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java:105]: Question: can we remove this Thread.sleep by reducing the delete interval for this test alone? is that configurable? And then by the time line 108 finishes, we can either assert or assertEventually on completedFiles size being 0

Comment (dada-dell-emc) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java:105]: The min value is 1 minute, which I seted up for these tests.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java (hunk 1 above)
Comment (dada-dell-emc) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java]: reduced sleep interval to 5 seconds from 50 seconds.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java (hunk 3 above)
Comment (dada-dell-emc) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java]: reduced sleep interval to 12-15 seconds from 25 seconds.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java (hunk 5 above)
Comment (dada-dell-emc) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java]: reduced sleep interval to 5 seconds from 50 seconds.

File: pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java (hunk 2 above)
Comment (dada-dell-emc) [reply pravega-sensor-collector/src/test/java/io/pravega/sensor/collector/PravegaSensorCollectorIntegrationTests.java]: removed order

**Review by sachin-j-joshi — APPROVED** [review]
LGTM

File: pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileIngestService.java [hunk 6]
```diff
@@ -137,8 +137,8 @@ long getMinTimeInMillisToUpdateFile() {
         return Long.parseLong(getProperty(MIN_TIME_IN_MILLIS_TO_UPDATE_FILE_KEY, "5000"));
     }
 
-    long getDeleteCompletedFilesIntervalInMinutes() {
-        return Long.parseLong(getProperty(DELETE_COMPLETED_FILES_INTERVAL_IN_MINUTES_KEY, "720"));
+    long getDeleteCompletedFilesIntervalInSeconds() {
+        return Long.parseLong(getProperty(DELETE_COMPLETED_FILES_INTERVAL_IN_SECONDS_KEY, "43200"));
```
Comment (krishan-Rai) [inline pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileIngestService.java:141]: is it like are we trying to delete completed file every 12 hours? @dada-dell-emc

File: gradle.properties [hunk 7]
```diff
@@ -13,11 +13,12 @@ commonsCLIVersion=1.4
 commonsCSVVersion=1.8
 commonsCodecVersion=1.14
 commonsMath3Version=3.6.1
-grizzlyVersion=3.1.3
+grizzlyVersion=2.35
```
Comment (krishan-Rai) [inline gradle.properties:16]: why we are going to lower version here  @dada-dell-emc  ?

File: pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileIngestService.java (hunk 6 above)
Comment (dada-dell-emc) [reply pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileIngestService.java:141]: Yes

File: gradle.properties (hunk 7 above)
Comment (dada-dell-emc) [reply gradle.properties:16]: Because of higher version there is conflicting versions dependency on glassfish library.

**Review by abhinb — APPROVED** [review]
LGTM

**Pull request merged**